### PR TITLE
Add factory context arg to http cache factory

### PR DESCRIPTION
**Pull request by EItanya**
Commit Message: Add factory context arg to http cache factory
Additional Description: Http Cache factory needs access to factory context to bootstrap async clients, and child stats scopes.
Risk Level: L
Testing: N/A
Docs Changes: N/A
Release Notes: N/A
Platform Specific Features: N/A

### CODE REVIEW

File: source/extensions/filters/http/cache/simple_http_cache/simple_http_cache.cc [hunk 1]
```diff
@@ -294,7 +294,8 @@ class SimpleHttpCacheFactory : public HttpCacheFactory {
         envoy::extensions::cache::simple_http_cache::v3::SimpleHttpCacheConfig>();
   }
   // From HttpCacheFactory
-  HttpCache& getCache(const envoy::extensions::filters::http::cache::v3::CacheConfig&) override {
+  HttpCache& getCache(const envoy::extensions::filters::http::cache::v3::CacheConfig&,
+                      Server::Configuration::FactoryContext&) override {
```
Comment (jmarantz) [inline source/extensions/filters/http/cache/simple_http_cache/simple_http_cache.cc:298]: It looks like you are plumbing in the context but then dropping it on the floo

I assume a follow-up PR will depend on this? Maybe comment in the description?

**Review by jmarantz — COMMENTED** [review]
basically looks good. need to fix format.

See https://github.com/envoyproxy/envoy/blob/main/CONTRIBUTING.md -- please don't rebase/force-push in envoy github PRs.

File: source/extensions/filters/http/cache/simple_http_cache/simple_http_cache.cc (hunk 1 above)
Comment (EItanya) [reply source/extensions/filters/http/cache/simple_http_cache/simple_http_cache.cc:298]: Fair point, I am building an extension in a separate codebase which relies on the async client factory, as well as the stats scope. I can add that in a comment.

**Comment by EItanya** [comment]
> please don't rebase/force-push in envoy github PRs.

Sounds good, I did that because the DCO check instructions mentioned said to do that.

**Comment by jmarantz** [comment]
yeah if you fail DCO you have to do that (or drop the PR and start a new one).

File: source/extensions/filters/http/cache/simple_http_cache/simple_http_cache.cc (hunk 1 above)
Comment (EItanya) [reply source/extensions/filters/http/cache/simple_http_cache/simple_http_cache.cc:298]: The context will be used for further extensions, not specifically for the in-memory cache. I can add a comment to the `HttpCacheFactory`

**Review by jmarantz — APPROVED** [review]

**Pull request merged**